packages/eslint-plugin/tests/rules/no-misused-promises.test.ts

```diff
@@ -304,6 +304,18 @@ declare const it: ItLike;
 it('', async () => {});
       `,
     },
+    {
+      code: `
+interface Props {
+  onEvent: (() => void) | (() => Promise<void>);
+}
+
+const Component: React.FC<Props> = () => null;
```

Comment:
nit we don't need to do any value magic here and we don't need to explicitly depend on the react types

```suggestion
declare function Component(props: Props): any;
```